Documentation/project-docs/platform-names.md

```diff
@@ -4,6 +4,24 @@ Over the years, we've accumulated quite a few names, platforms, and components t
 
 This will also list some aliases. As you'll see the aliases aren't always correct -- they are merely listed to help you find the better and less confusing terminology.
 
+In this document, the following terms are used:
+
+* **IL**. Intermediate language. IL is a stack-based instruction set and is what higher level languages, such as C#, compile down to.
+
+* **GC**. Garbage collector. Garbage collection is an implementation of automatic memory management. .NET uses a generational garbage collector, i.e. it groups objects into generations to limit the number of nodes it has to walk for determining which objects are alive. This speeds up collection times.
+
+* **JIT**. Just in time compiler. This technology compiles IL to machine code that the processor understands. It's called JIT because compilation happens on demand and is performed on the same machine the code needs to run on. Since JIT compilation happens during execution of the application, compile time is part of the run time. Thus, JIT compilers have to trade spending more time optimizing code with the savings the resulting code can produce.
+
+* **AOT**. Ahead of time compiler. Similar to JIT this compiler also translates IL to machine code. However, in contrast to a JIT the compilation happens before the application is executed and is usually performed on a different machine. AOT tool chains don't have to trade run time and compile time and thus can spend more time optimizing. Also, since the context of AOT are entire applications, AOT compiler can also perform cross module linking and whole program analysis.
+
+* **NGEN**. Native (image) generation. You can think of this technology as a persistent JIT compiler. It usually compiles code on the same machine it's supposed to run on, but typically happens during install time.
+
+* **CLR**. Common language runtime. Exact meaning depends on context, but it usually refers to the runtime of the .NET Framework and includes several components. The CLR is a virtual machine, i.e. it includes the facilities to generate and compile code on the fly using a JIT.
+
+* **CoreCLR**. Core Common Language Runtime. Built from the same code base as the CLR. Originally, CoreCLR was the runtime of Silverlight but is now part of .NET Core. You can think of CoreCLR as a simplified version of the CLR: it's also a virtual machine with a JIT.
+
+* **CoreRT**. Core runtime. In contrast to the CLR/CoreCLR, CoreRT is not a virtual machine, i.e. doesn't include the facilities to generate and run code on the fly because it doesn't include a JIT. It does, however, include the GC and the ability for runtime type identification (RTTI) as well as reflection. However, it's type system is designed so that metadata for reflection can be omitted. This enables having an AOT tool chain that can link away superfluous metadata (and more importantly) identify code that the application doesn't use.
+
```

Comment:
I think I'd add something about how CLR is Windows only vs CoreCLR and CoreRT being cross-platform